### PR TITLE
No issue: Fix leaks in Home and Bookmarks

### PR DESCRIPTION
**Pull request by ekager**
Screenshot_20200417-182847

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by ekager** [comment]
Screenshot_20200417-192110

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -356,6 +359,7 @@ class HomeFragment : Fragment() {
 
     override fun onDestroyView() {
         super.onDestroyView()
+        _sessionControlInteractor = null
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:362]: Interesting, I guess this helps to GC this earlier - I wonder if it's fair to call this a leak in a true sense. Once HomeFragment goes through `onCreateView` again, the old instance of sessionControlInteractor will no longer be referenced, and it will get GC'd. But, that won't happen _until_ user goes back to HomeFragment, so we are retaining this stuff in memory for longer than necessary.

You _could_ similarly null more things here, `currentMode`, `homeFragmentStore`, etc - same logic applies.

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt [hunk 2]
```diff
@@ -298,6 +300,11 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
         )
     }
 
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _bookmarkInteractor = null
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt:305]: Why is this needed?

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt (hunk 1 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:362]: The main problem with these two objects specifically is that it they retains references to the fragment views that are being destroyed so they are leaking and this can't be GC I believe. Here specifically we are passing in a lambda that references the fragment view and in bookmarks, we pass in a view reference for the snackbar.

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:362]: https://stackoverflow.com/a/59504797

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt (hunk 2 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt:300]: Here's a second possible fix to the same leak: https://github.com/mozilla-mobile/fenix/compare/master...ekager:bookmark-snackbar-fix2

Maybe slightly preferable since I really don't think the interactor should have references to any views to begin with... But also always leaves the possibility that someone else adds a reference later and regresses that fix. The definitive nulling out of the interactor protects against that. 🤷‍♀️
